truss/remote/baseten/tar.py

```diff
@@ -0,0 +1,50 @@
+import tarfile
+import tempfile
+from pathlib import Path
+from typing import IO, Any, Callable
+
+from tqdm import tqdm
+
+
+class ReadProgressIndicatorFileHandle:
+    def __init__(
+        self, file: IO[bytes], progress_callback: Callable[[int], Any]
+    ) -> None:
+        self._file = file
+        self._progress_callback = progress_callback
+
+    def read(self, size=-1):
+        data = self._file.read(size)
+        self._progress_callback(len(data))
+        return data
+
+    def __getattr__(self, attr):
+        return getattr(self._file, attr)
+
+
+def create_tar_with_progress_bar(source_dir: Path, delete=True):
+    total_size = sum(f.stat().st_size for f in source_dir.glob("**/*") if f.is_file())
+
+    # Keeping the .tgz suffix for backwards compatibility even though
+    # this tar is uncompressed for upload
+    temp_file = tempfile.NamedTemporaryFile(suffix=".tgz", delete=delete)
```

Comment:
can you test if we still need .tgz? I don't recall. I think we do.

Reply:
It seems like it worked with .zip as well. But I'm inclined to keep it this way, not sure if there are 2nd order effects to changing this.